Remove page size hint from options objects in digital twins APIs

timtay-microsoft opened the pull request:
Users are expected to provide the page size hint in the .AsPages() methods in the pageable types themselves

vinagesh commented on sdk/digitaltwins/Azure.DigitalTwins.Core/tests/QueryTests.cs:
```diff
@@ -124,7 +121,7 @@ await foreach (Page<string> queriedTwinPage in asyncPageableResponse.AsPages())
                 // elements, or have no continuation token (signaling that it is the last page)
                 int pageCount = 0;
-                await foreach (Page<string> page in client.QueryAsync(queryString, options).AsPages())
+                await foreach (Page<string> page in client.QueryAsync(queryString).AsPages(null, pageSize))
```
I would suggest using named parameters here so that we can clearly see that "continuationToken" is being set to null.

vinagesh commented on sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/Models/GetDigitalTwinsEventRouteOptions.cs:
```diff
@@ -18,5 +18,8 @@ public partial class GetDigitalTwinsEventRouteOptions
         [CodeGenMember("Tracestate")]
         public string TraceState { get; set; }
+
+        // This reduces the scope of this property to private since users shouldn't set page size here, but on the pageable instance's .AsPages() method instead.
```
This could be worded - "This is internal because users should not set page size here. It should be set on the pageable instances's .AsPages() method."

timtay-microsoft committed:
Remove page size hint from options objects in digital twins APIs

Users are expected to provide the page size hint in the .AsPages() methods in the pageable types themselves

timtay-microsoft committed:
record tests
